src/pmi/src/pmi_v1.c

```diff
@@ -760,6 +760,14 @@ static int PMII_Set_from_port(int id)
     pmi_errno = PMIU_cmd_get_response(PMI_fd, &pmicmd);
     PMIU_ERR_POP(pmi_errno);
 
+    const char *spawner_jobid;
+    int appnum, verbose;        /* unused */
+    PMIU_msg_get_response_fullinit(&pmicmd, &PMI_rank, &PMI_size, &appnum, &spawner_jobid,
+                                   &verbose);
+    PMIU_ERR_POP(pmi_errno);
+
+    PMI_spawned = (spawner_jobid != NULL);
+
```

Comment:
Only the new hydra sets all these fields in the `initack` response. Old `hydra` relies on the 3 extra `set` command to set `size/rank/debug`, which is handled in `PMIU_cmd_get_response` already. Thus, if used with an old `hydra` or a third party PMI server, the extra `PMIU_msg_get_response_fullinit` may fail or overwrite the set values.

Reply:
Just tested it with old `hydra`, it returns `PMI_FAIL` with `PMIU_msg_get_response_fullinit` due to missing fields in the `initack` message.

For backward compatibility, we should not add this patch.

Reply:
Maybe we should add a test using old `hydra` to make sure we do not break backward compatibility.

Reply:
Oh, I see. What a waste of time that was 😦.

Reply:
I'm going to move the size/rank/debug settings into pmi_v1.c because the fullinit abstraction is broken and misleading.